build_web_compilers/lib/src/ddc_names.dart

```diff
@@ -2,10 +2,25 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// Escape [name] to make it into a valid identifier.
+import 'package:path/path.dart' as _p;
+
+/// Transforms a path to a valid JS identifier.
 ///
-/// This is copied from
-/// https://github.com/dart-lang/sdk/blob/300fd66f5c5e7497674ea4dd9fd21be91ee3c71e/pkg/dev_compiler/lib/src/analyzer/code_generator.dart#L6390
+/// This logic must be synchronized with [pathToJSIdentifier] in DDC at:
+/// pkg/dev_compiler/lib/src/compiler/module_builder.dart
+String pathToJSIdentifier(String path) {
+  path = _p.basenameWithoutExtension(_p.normalize(path));
+  if (path.startsWith('/') || path.startsWith('\\')) {
```

Comment:
When would we end up with `\` at the start?

Reply:
I think this is more a side effect of stuff happening on DDC's side. I thought it better to keep the code consistent (even if there's some extraneous logic).